cider/tests/utils/CiderBatchChecker.cpp

```diff
@@ -144,6 +144,47 @@ bool CiderBatchChecker::checkOneScalarBatchEqual(const ScalarBatch<T>* expected_
   return true;
 }
 
+template <>
+bool CiderBatchChecker::checkOneScalarBatchEqual<bool>(
+    const ScalarBatch<bool>* expected_batch,
+    const ScalarBatch<bool>* actual_batch) {
+  auto expected_data_buffer = expected_batch->getRawData();
+  auto expected_validity_buffer = expected_batch->getNulls();
+  auto actual_data_buffer = actual_batch->getRawData();
+  auto actual_validity_buffer = actual_batch->getNulls();
+
+  // both validity buffer and data buffer are bitmaps
+  // we compare both of them in one step with a bitwise AND
+  auto row_num = actual_batch->getLength();
+  auto bytes = ((row_num + 7) >> 3);
+
+  for (int i = 0; i < bytes; ++i) {
+    // apply bitwise AND
+    uint8_t expected_masked = expected_validity_buffer
+                                  ? expected_data_buffer[i] & expected_validity_buffer[i]
+                                  : expected_data_buffer[i];
+    uint8_t actual_masked = actual_validity_buffer
+                                ? actual_data_buffer[i] & actual_validity_buffer[i]
+                                : actual_data_buffer[i];
+
+    // for the last byte, clear trailing padding values to prevent false positives caused
+    // by differences in padding
+    if (i == bytes - 1) {
```

Comment:
can we move this if block out of for loop?

Reply:
Not sure if I got it right. This `if` block is like a preprocessing step for the last byte. If we moved this `if` out of the `for` loop, we would have to move the check for the last byte out of this loop.